lib/writer.js

```diff
@@ -84,11 +85,7 @@ Writer.prototype.publish = function(topic, msg, fn){
     }
   } else {
     // wait for ready and retry
-    var self = this;
-
-    this.once('ready', function(){
-      self.publish(topic, msg, fn)
-    });
+    this.publishQueue.push([topic, msg, fn]);
```

Comment:
above this, there is a `debug` message:

```    debug('%s - publish', conn.addr);```

should we also have a message for deferring messages, and include the queue depth?  it would give us better visibility into how often this is occurring and how urgent it could end up being.

Reply:
There are other debug messages that may be better suited to help with this issue without the risk of being very spammy.

```
  conn.on('close', function(){
    debug('%s - remove from pool', addr);
    ...
  });
  conn.on('ready', function(){
    debug('%s - add to pool', addr);
    ...
```

Maybe we could add the total amount of connections on these messages, that way we know when the pool is empty. We could also add an extra log on the `ready` event handler that would mention how many queued messages were added and are going to be published.

Reply:
> Maybe we could add the total amount of connections on these messages, that way we know when the pool is empty.

yeah, having at least the number of messages will help from a visibility perspective.  it's important to know how often this is occurring and what kind of impact it's having.

Reply:
Updated debugs.